### PR TITLE
Add i18n support

### PR DESCRIPTION
**Pull request by TheLastProject**
Adds global i18n support and builds the translations for the settings app. Currently non-functional. DO NOT MERGE.

In my idea, this should make i18n work. It at least puts all the files on the right spot with the correct rights. Unfortunately, after building and testing this, I got the following error on my device when trying to launch any app: `lipstick[850]: wl_event_loop_dispatch error: -1`. Why this happens is beyond me.

The build script also returns 2 QA warnings:
```
WARNING: QA Issue: python-pygobject: /python-pygobject/usr/lib/python2.7/site-packages/pygtk.pth is owned by uid 1000, which is the same as the user running bitbake. This may be due to host contamination [host-user-contaminated]
WARNING: QA Issue: asteroid-launcher: /asteroid-launcher/var/lib/environment/ceres/locale.conf is owned by uid 1000, which is the same as the user running bitbake. This may be due to host contamination [host-user-contaminated]
```

These warnings are caused by doing a chown 1000:1000 to make ceres the owner at a point of the build script where the user ceres does not exist yet.

This pull request is supposed to accompany https://github.com/AsteroidOS/asteroid-settings/pull/1. While this breaks app launching on my device, I really can't figure out why, so I'm making this pull request to simplify giving feedback for the time being so this issue can be resolved.

### CODE REVIEW

File: recipes-asteroid/asteroid-launcher/asteroid-launcher_git.bb [hunk 1]
```diff
@@ -23,4 +29,14 @@ do_install_append() {
     install -d ${D}/usr/lib/systemd/user/default.target.wants/
     cp ../asteroid-launcher.service ${D}/usr/lib/systemd/user/
     ln -s ../asteroid-launcher.service ${D}/usr/lib/systemd/user/default.target.wants/asteroid-launcher.service
+
+    install -d ${D}/etc/systemd/system/user@.service.d/
+    cp ../localeEnv.conf ${D}/etc/systemd/system/user@.service.d/locale.conf
+
+    install -d ${D}/var/lib/environment/ceres/
+    cp ../locale.conf ${D}/var/lib/environment/ceres/locale.conf
+    chown 1000:1000 ${D}/var/lib/environment/ceres/locale.conf # ceres:ceres
```
Comment (TheLastProject) [inline recipes-asteroid/asteroid-launcher/asteroid-launcher_git.bb]: This causes the QA warnings, but because the user ceres does not exist at this point, running `chown ceres:ceres` is not possible.

File: recipes-asteroid/asteroid-calendar/asteroid-calendar_git.bb [hunk 2]
```diff
@@ -10,4 +10,5 @@ PV = "+git${SRCREV}"
 S = "${WORKDIR}/git"
 inherit qmake5
 
+RDEPENDS_${PN} += "qtquickcontrols-qmlplugins"
```
Comment (FlorentRevest) [inline recipes-asteroid/asteroid-calendar/asteroid-calendar_git.bb]: This needs to be put inside a different commit. Internationalization is not the right place for that modif

File: recipes-asteroid/asteroid-launcher/asteroid-launcher/jolla-supported-languages/de_DE.conf [hunk 3]
```diff
@@ -0,0 +1,4 @@
+Name=Deutsch
+LocaleCode=de_DE.utf8
+Region=Deutschland
+RegionLabel=Region: %1
```
Comment (FlorentRevest) [inline recipes-asteroid/asteroid-launcher/asteroid-launcher/jolla-supported-languages/de_DE.conf]: Let's create a new supported-language recipe and get rid of the jolla prefix in nemo-qml-plugin-systemsettings

File: recipes-asteroid/asteroid-launcher/asteroid-launcher_git.bb (hunk 1 above)
Comment (FlorentRevest) [inline recipes-asteroid/asteroid-launcher/asteroid-launcher_git.bb]: It's bitbake's job to take care of file owner. Don't do this yourself !

File: recipes-asteroid/asteroid-launcher/asteroid-launcher_git.bb [hunk 4]
```diff
@@ -23,4 +29,14 @@ do_install_append() {
     install -d ${D}/usr/lib/systemd/user/default.target.wants/
     cp ../asteroid-launcher.service ${D}/usr/lib/systemd/user/
     ln -s ../asteroid-launcher.service ${D}/usr/lib/systemd/user/default.target.wants/asteroid-launcher.service
+
+    install -d ${D}/etc/systemd/system/user@.service.d/
+    cp ../localeEnv.conf ${D}/etc/systemd/system/user@.service.d/locale.conf
```
Comment (FlorentRevest) [inline recipes-asteroid/asteroid-launcher/asteroid-launcher_git.bb]: This file should be installed in the new supported languages recipe. It also has nothing to do here

File: recipes-asteroid/asteroid-settings/asteroid-settings_git.bb [hunk 5]
```diff
@@ -11,4 +11,11 @@ S = "${WORKDIR}/git"
...
 
 DEPENDS += "qml-asteroid nemo-qml-plugin-systemsettings nemo-qml-plugin-dbus qtconnectivity mapplauncherd-booster-qtcomponents"
-RDEPENDS_${PN} += "qtconnectivity-qmlplugins nemo-qml-plugin-systemsettings nemo-qml-plugin-dbus"
+RDEPENDS_${PN} += "qtconnectivity-qmlplugins nemo-qml-plugin-systemsettings nemo-qml-plugin-dbus qtquickcontrols-qmlplugins"
+FILES_${PN} += "/usr/share/translations/"
+
+do_install_append() {
+    lrelease ${WORKDIR}/git/asteroid-settings.pro
+    install -d ${D}/usr/share/translations/
+    cp ${WORKDIR}/git/asteroid-settings.*.qm ${D}/usr/share/translations/
```
Comment (FlorentRevest) [inline recipes-asteroid/asteroid-settings/asteroid-settings_git.bb]: Can't you do all of thus from asteroid-settings.pro  ?

File: recipes-asteroid/asteroid-launcher/asteroid-launcher_git.bb (hunk 1 above)
Comment (TheLastProject) [inline recipes-asteroid/asteroid-launcher/asteroid-launcher_git.bb]: The issue is that bitbake creates this file as root, which makes the asteroid-settings app unable to write the selected language to file. However, changing the permissions to ceres is probably wrong anyway, as any app could set environment variables then. Is there some way to only give asteroid-settings write access to this file?

File: recipes-asteroid/asteroid-calendar/asteroid-calendar_git.bb (hunk 2 above)
Comment (TheLastProject) [inline recipes-asteroid/asteroid-calendar/asteroid-calendar_git.bb]: Fair point, will do that.

File: recipes-asteroid/asteroid-launcher/asteroid-launcher/jolla-supported-languages/de_DE.conf (hunk 3 above)
Comment (TheLastProject) [inline recipes-asteroid/asteroid-launcher/asteroid-launcher/jolla-supported-languages/de_DE.conf]: Okay, will do.

File: recipes-asteroid/asteroid-launcher/asteroid-launcher_git.bb (hunk 4 above)
Comment (TheLastProject) [inline recipes-asteroid/asteroid-launcher/asteroid-launcher_git.bb]: Will do.

File: recipes-asteroid/asteroid-settings/asteroid-settings_git.bb (hunk 5 above)
Comment (TheLastProject) [inline recipes-asteroid/asteroid-settings/asteroid-settings_git.bb]: Languages aren't picked up by the normal Qt compiler, we specifically need to run lrelease to create the "compiled language files". If you mean the install -d and cp part, I don't think that can be set up in asteroid-settings.pro. Which files to compile from .ts to .qm however, is defined in asteroid-settings.pro. See https://github.com/TheLastProject/asteroid-settings/blob/8066edab1197269e8d0e10d15e1251e0482976f4/asteroid-settings.pro#L21-L36 for that.

File: recipes-asteroid/supported-language/supported-language_git.bb [hunk 6]
```diff
@@ -0,0 +1,25 @@
+SUMMARY = "Asteroid's launcher based on lipstick"
```
Comment (FlorentRevest) [inline recipes-asteroid/supported-language/supported-language_git.bb]: Are you sure? :)

Comment (TheLastProject) [inline recipes-asteroid/supported-language/supported-language_git.bb]: Whoops! Just a minute!

**Comment by TheLastProject** [comment]
After making your suggested changes, there are now 4 warnings (of which 3 are that Bitbake doesn't seem to recognise CC0):
```
WARNING: supported-languages: No generic license file exists for: CC0 in any provider
WARNING: QA Issue: supported-languages: /supported-languages/var/lib/environment/ceres/locale.conf is owned by uid 1000, which is the same as the user running bitbake. This may be due to host contamination [host-user-contaminated]
WARNING: The license listed CC0 was not in the licenses collected for recipe supported-languages
WARNING: [log_check] asteroid-image: found a warning message in the logfile (keyword 'WARNING:'):
[log_check] WARNING: The license listed CC0 was not in the licenses collected for recipe supported-languages
```

Despite these warning, everything seems to work flawlessly.
